app/javascript/mastodon/components/media_gallery.js

```diff
@@ -162,12 +164,13 @@ class Item extends React.PureComponent {
 
       thumbnail = (
         <a
-          className='media-gallery__item-thumbnail'
+          className={`media-gallery__item-thumbnail ${!attachment.get('description') && 'media-missing-description'}`}
           href={attachment.get('remote_url') || originalUrl}
           onClick={this.handleClick}
           target='_blank'
           rel='noopener noreferrer'
         >
+          { !attachment.get('description') && <IconButton className='media-gallery__item-no-description media__no-description-icon' title={this.props.noDescriptionTitle} icon='exclamation-triangle' style={{ right: `calc(4px + ${(left === 'auto' ? '0px' : left)})`, bottom: `calc(4px + ${(top === 'auto' ? '0px' : top)})` }} overlay /> }
```

Comment:
I'm not super happy with hardcoding style values like this, but it's consistent with what's done in that class overall, so I'm fine with it.